### PR TITLE
[BPK-1644] Fix iOS 11 + RN 0.55.X font weight issue

### PR DESCRIPTION
**Pull request by matthewdavidson**
Introduces an iOS 11 + React Native 0.55.X specific fix for `BpkText` 

`font-weight` rendering of system fonts (San Francisco).

**The issue:**

Default | Emphasize
--- | ---
before | before emphasize

**The fix:**

Default | Emphasize
--- | ---
after | after emphasize

### CODE REVIEW

File: packages/bpk-tokens/tokens/base.raw.ios.json [hunk 1]
```diff
@@ -87,9 +87,6 @@
     "FONT_WEIGHT_BOLD": {
       "value": "700"
     },
-    "FONT_WEIGHT_HEAVY": {
```
Comment (matthewdavidson) [inline packages/bpk-tokens/tokens/base.raw.ios.json]: This was unused - aliases are not included in output (apart from raw but these token formats are not "advertised") so non breaking change.

File: packages/bpk-tokens/tokens/base.raw.android.json [hunk 2]
```diff
@@ -102,12 +102,6 @@
     "FONT_SIZE_XXL": {
       "value": 34
     },
-    "FONT_WEIGHT_BLACK": {
-      "value": "900"
-    },
-    "FONT_WEIGHT_BOLD": {
-      "value": "700"
-    },
```
Comment (matthewdavidson) [inline packages/bpk-tokens/tokens/base.raw.android.json]: These were unused - aliases are not included in output (apart from raw but these token formats are not "advertised") so non breaking change.

File: packages/bpk-tokens/src/ios/base/aliases.json [hunk 3]
```diff
@@ -20,7 +20,6 @@
     "FONT_WEIGHT_NORMAL": "400",
     "FONT_WEIGHT_SEMIBOLD": "600",
     "FONT_WEIGHT_BOLD": "700",
-    "FONT_WEIGHT_HEAVY": "800",
```
Comment (matthewdavidson) [inline packages/bpk-tokens/src/ios/base/aliases.json]: This was unused - aliases are not included in output (apart from raw but these token formats are not "advertised") so non breaking change.

File: packages/bpk-tokens/src/android/base/aliases.json [hunk 4]
```diff
@@ -20,8 +20,6 @@
     "LINE_HEIGHT_XXL": 40,
     "FONT_WEIGHT_REGULAR": "400",
     "FONT_WEIGHT_MEDIUM": "500",
-    "FONT_WEIGHT_BOLD": "700",
-    "FONT_WEIGHT_BLACK": "900",
```
Comment (matthewdavidson) [inline packages/bpk-tokens/src/android/base/aliases.json]: These were unused - aliases are not included in output (apart from raw but these token formats are not "advertised") so non breaking change.

File: native/packages/react-native-bpk-component-text/stories.js [hunk 5]
```diff
@@ -14,7 +34,7 @@ import BpkText from './index';
 
 storiesOf('react-native-bpk-component-text', module)
   .addDecorator(CenterDecorator)
-  .add('docs:text-styles', () => (
+  .add('docs:default', () => (
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-text/stories.js]: Aligning to convention

File: native/packages/react-native-bpk-component-text/stories.js [hunk 6]
```diff
@@ -24,7 +44,7 @@ storiesOf('react-native-bpk-component-text', module)
       <BpkText textStyle="xs">Flights to Edinburgh</BpkText>
     </View>
   ))
-  .add('Bold', () => (
+  .add('docs:emphasize', () => (
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-text/stories.js]: Aligning to convention

File: native/packages/react-native-bpk-component-text/src/BpkText-test.common.js [hunk 7]
```diff
@@ -95,8 +101,11 @@ const commonTests = () => {
         BpkText.propTypes
           .emphasize(
             {
+              children: 'Lorem ipsum.',
               textStyle: 'sm',
+              // $FlowFixMe
```
Comment (matthewdavidson) [inline native/packages/react-native-bpk-component-text/src/BpkText-test.common.js]: This test explicitly checks for the wrong type being passed - perhaps we can just remove it?

File: native/packages/react-native-bpk-component-text/src/BpkText.js [hunk 8]
```diff
@@ -94,40 +94,38 @@ const styles = StyleSheet.create({
...
+      style.push({ fontWeight: '700' });
+    }
   }
 
-  finalStyle.push(style);
+  if (shouldApplyFontWeightFix && textStyle === 'xxl') {
+    style.push({ fontWeight: '800' });
+  }
+
+  style.push(userStyle);
```
Comment (k0nserv) [inline native/packages/react-native-bpk-component-text/src/BpkText.js]: This should only be done if there actually is a `userStyle` imo.

File: native/packages/react-native-bpk-component-text/src/BpkText.js [hunk 9]
```diff
@@ -94,40 +94,38 @@ const styles = StyleSheet.create({
...
   // Emphasize on iOS is not supported for the XXL size. This is also checked with
   // the `emphasizePropType` prop.
   const shouldEmpasize =
     emphasize && !(Platform.OS === 'ios' && textStyle === 'xxl');
 
   if (shouldEmpasize) {
-    finalStyle.push(getEmphasizeProperties(props));
+    style.push(getEmphasizeProperties());
+
+    if (shouldApplyFontWeightFix) {
```
[the remark this answers is not in the document]
Comment (matthewdavidson) [reply native/packages/react-native-bpk-component-text/src/BpkText.js]: resolved offline - there is already a comment in `font-weight-fix.js`.

**Pull request merged**